ksql-engine/src/main/java/io/confluent/ksql/function/udf/string/Substring.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.function.udf.string;
+
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+
+@UdfDescription(name = "substring",
+    author = "Confluent",
+    description = "returns a substring of the passed in value")
+public class Substring {
+
+  @Udf(description = "Returns a string that is a substring of this string. The"
```

Comment:
These look great! This text will make it into the self documenting `SHOW FUNCTIONS` and `HELP FUNCTION substring` system that we build, right?

Reply:
Yes that is the plan